### PR TITLE
Remove unused patches/ files as already in vendor-patches directory and use NewOptimizedDirectorySourceLocator

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Review by TomasVotruba — APPROVED** [review]

**Comment by samsonasik** [comment]
New PHPStan notice seems due to new PHPStan release:

https://github.com/phpstan/phpstan/releases/tag/1.10.31

```

Run vendor/bin/phpstan analyse --ansi --error-format symplify
Note: Using configuration file /home/runner/work/rector-src/rector-src/phpstan.neon.
    0/2091 [░░░░░░░░░░░░░░░░░░░░░░░░░░░░]   0%
  420/2091 [▓▓▓▓▓░░░░░░░░░░░░░░░░░░░░░░░]  20%
 1060/2091 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░░░░░░░░░░░]  50%
 2091/2091 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------------------------------------------------------------------------
  packages/NodeTypeResolver/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php:68
 ------------------------------------------------------------------------
  - '#Instantiation of deprecated class PHPStan\\Reflection\\BetterReflection\\SourceLocator\\OptimizedDirectorySourceLocator\:
Use NewOptimizedDirectorySourceLocator#'
 ------------------------------------------------------------------------
```

**Comment by samsonasik** [comment]
Creation service may from `OptimizedDirectorySourceLocatorFactory`, I will check.

**Comment by samsonasik** [comment]
@TomasVotruba let's try using `OptimizedDirectorySourceLocatorFactory` https://github.com/rectorphp/rector-src/pull/4844/commits/91bcb7c5cea9c8ff8e997fefc89fd9b53345e03c

**Comment by samsonasik** [comment]
Bump PHPStan to 1.10.31 to ensure `OptimizedDirectorySourceLocatorFactory` exists 👍  https://github.com/rectorphp/rector-src/pull/4844/commits/f68bd9ce79e0467f92feb5eecfb530d17fddf764

**Comment by samsonasik** [comment]
`filesByDirectory` seems not needed, use `directories` instead and let `OptimizedDirectorySourceLocatorFactory` locate it.

**Comment by samsonasik** [comment]
@TomasVotruba I removed `PhpFilesFinder` which seems the filter `.blade.php` only on directory, while not on file, which seems inconsistent cdd9fa0425896f408291cb93dfaf877b94d5eb26

**Comment by samsonasik** [comment]
It seems cause error:

```
➜  rector-src git:(remove-unused-patches) vendor/bin/phpunit packages-tests/Skipper/Skipper/SkipperRectorRuleTest.php
PHPUnit 10.3.2 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.5
Configuration: /Users/samsonasik/www/rector-src/phpunit.xml

F                                                                   1 / 1 (100%)

Time: 00:00.023, Memory: 20.00 MB

There was 1 failure:

1) Rector\Tests\Skipper\Skipper\SkipperRectorRuleTest::testRemovingServiceFromContainer
Failed asserting that actual size 2 matches expected size 1.
```

File: packages-tests/Skipper/Skipper/SkipperRectorRuleTest.php [hunk 1]
```diff
@@ -34,7 +33,7 @@ public function testRemovingServiceFromContainer(): void
         $container = self::getContainer();
 
         // to invoke before resolving
-        $container->make(PhpFilesFinder::class);
+        $container->make(\PHPStan\Reflection\BetterReflection\SourceLocator\FileNodesFetcher::class);
```
Comment (samsonasik) [inline packages-tests/Skipper/Skipper/SkipperRectorRuleTest.php]: this seems magic just to invoke to hook the service

**Comment by samsonasik** [comment]
Should be fixed now 👍

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba let's give it a try 👍

**Comment by samsonasik** [comment]
@TomasVotruba it seems cause error in downgrade build:

https://github.com/rectorphp/rector-src/actions/runs/5965685296/job/16183665877#step:10:19

I will revert `NewOptimizedDirectorySourceLocator` to only remove `patches` dir.

**Comment by samsonasik** [comment]
I created PR:

- https://github.com/rectorphp/rector-src/pull/4845

to revert use NewOptimizedDirectorySourceLocator

**Comment by TomasVotruba** [comment]
These should be 2 separated PRs, ideally with PHPStan lock in patch removing instead of fixing and removing at once.

**Pull request merged**